benchmarks/inference-server/text-generation-inference/custom-metrics-stackdriver-adapter/main.tf

```diff
@@ -1,278 +0,0 @@
-resource "kubernetes_namespace_v1" "custom-metrics" {
-  metadata {
-    name = "custom-metrics"
-  }
-}
-
-resource "kubernetes_service_account_v1" "custom-metrics-stackdriver-adapter-no-wi" {
-  count = var.workload_identity.enabled ? 0 : 1
-  metadata {
-    name      = "custom-metrics-stackdriver-adapter"
-    namespace = kubernetes_namespace_v1.custom-metrics.metadata[0].name
-  }
-}
-
-resource "kubernetes_service_account_v1" "custom-metrics-stackdriver-adapter-wi" {
-  count = var.workload_identity.enabled ? 1 : 0
-  metadata {
-    name      = "custom-metrics-stackdriver-adapter"
-    namespace = kubernetes_namespace_v1.custom-metrics.metadata[0].name
-    annotations = {
-      "iam.gke.io/gcp-service-account" = google_service_account.cmsa-sa[0].email
-    }
-  }
-}
-
-resource "kubernetes_cluster_role_binding_v1" "custom-metrics-system-auth-delegator" {
-  metadata {
-    name = "custom-metrics:system:auth-delegator"
-  }
-  role_ref {
-    api_group = "rbac.authorization.k8s.io"
-    kind      = "ClusterRole"
-    name      = "system:auth-delegator"
-  }
-  subject {
-    kind = "ServiceAccount"
-    name = (var.workload_identity.enabled
```

Comment:
NB: The replacement (rolebinding_custom-metrics-auth-reader.yaml.tftpl) hard-codes the service account name. i.e. here: https://github.com/GoogleCloudPlatform/ai-on-gke/blob/f2883eb600045e9614726c8bf28123dca15885d2/modules/custom-metrics-stackdriver-adapter/templates/rolebinding_custom-metrics-auth-reader.yaml.tftpl#L12

Technically, that means that terraform doesn't know that it should create the service account before the role binding. (Probably, terraform just executes blocks in order if there's no dependency, so I suspect it works anyways.)

To fix that, you could parameterize the `*.tftpl` files to enforce correct ordering.

This is unrelated to this PR though, as that ordering concern pre-exists.